### PR TITLE
Enable generation of arbitrary annotations and test their equivalence

### PR DESCRIPTION
**Pull request by shaobo-he-aws**
*Issue #, if available:*

*Description of changes:*

Depend on https://github.com/cedar-policy/cedar/pull/1382

### CODE REVIEW

File: cedar-drt/fuzz/src/schemas.rs [hunk 1]
```diff
@@ -88,13 +88,40 @@ impl<'a, T: Equiv> Equiv for &'a T {
...
+                }
+            }
+            (Some(a), None) | (None, Some(a)) => {
+                if a.val.is_empty() {
+                    return Ok(());
+                }
+            }
+            (None, None) => return Ok(()),
+        };
+        Err(format!("two annotations mismatch: {lhs:?} and {rhs:?}"))
```
Comment (adpaco-aws) [inline cedar-drt/fuzz/src/schemas.rs:112]:
```suggestion
        Err(format!("{lhs:?} and {rhs:?} are not equivalent"))
```

Comment (adpaco-aws) [reply cedar-drt/fuzz/src/schemas.rs:112]: The error is already wrapped in `mismatch in namespace annotations: {e}` so there's no need to repeat the description.

File: cedar-drt/fuzz/src/schemas.rs [hunk 2]
```diff
@@ -517,3 +552,70 @@ impl Equiv for cedar_policy_validator::ValidatorSchema {
...
+                serde_json::from_value(serde_json::json!({"b": "🥯🍩", "a": "🥨"})).unwrap(),
+            ),
+        ];
+        pairs
+            .iter()
+            .for_each(|(a, b)| assert!(Equiv::equiv(a, b).is_ok()));
+
+        // negative cases
+        let pairs: [(Annotations, Annotations); 5] = [
+            (
```
Comment (adpaco-aws) [inline cedar-drt/fuzz/src/schemas.rs:596]: Nit: we could have this be a separate `#[test]`

**Review by john-h-kastner-aws — APPROVED** [review]

File: cedar-drt/fuzz/src/schemas.rs (hunk 2 above)
Comment (shaobo-he-aws) [reply cedar-drt/fuzz/src/schemas.rs:596]: Will do in another PR.

**Review by adpaco-aws — APPROVED** [review]

**Pull request merged**